Add production checklist about `maxNumEventLoopsPerEndpoint` (#5297)
line/armeria: site/src/pages/docs/advanced-production-checklist.mdx

  sb.childChannelOption(ChannelOption.SO_RCVBUF, ...);

  // Client-side
  ClientFactoryBuilder cfb = ClientFactory.builder();
  cfb.channelOption(ChannelOption.SO_REUSEADDR, ...);
  cfb.channelOption(ChannelOption.SO_SNDBUF, ...);
  cfb.channelOption(ChannelOption.SO_RCVBUF, ...);
  ClientFactory cf = cfb.build();
  ClientBuilder cb = Clients.builder(...);
  cb.factory(cf);
  ```

> - Consider increasing <type://ClientFactoryBuilder.maxNumEventLoopsPerEndpoint(int)>
>   and <type://ClientFactoryBuilder.maxNumEventLoopsPerHttp1Endpoint(int)> when a

Reviewer: ```suggestion
- Consider increasing <type://ClientFactoryBuilder#maxNumEventLoopsPerEndpoint(int)>
  and <type://ClientFactoryBuilder#maxNumEventLoopsPerHttp1Endpoint(int)> when a
```

> - Consider increasing `<type://ClientFactoryBuilder.maxNumEventLoopsPerEndpoint(int)>`
>   and `<type://ClientFactoryBuilder.maxNumEventLoopsPerHttp1Endpoint(int)>` when a

Reviewer: nit: No need to surround the type using ``` ` ``` 
```suggestion
- Consider increasing <type://ClientFactoryBuilder.maxNumEventLoopsPerEndpoint(int)>
  and <type://ClientFactoryBuilder.maxNumEventLoopsPerHttp1Endpoint(int)> when a
```

  client needs to send a large number of requests to a specific endpoint. The client
  will assign more CPU resources and create more connections by increasing the number

>   of event loops.

Reviewer: How about 
```suggestion
  of event loops up to the quantity in <type://ClientFactory#eventLoopGroup()>.
```

  ClientBuilder cb = Clients.builder(...);
  cb.factory(cf);
  ```

> - Increase `maxNumEventLoopsPerEndpoint` and `maxNumEventLoopsPerHttp1Endpoint` when your clients send requests

Reviewer 2: How about explaining the effect of increasing the number of event loops?
```suggestion
- Increase `<type://ClientFactoryBuilder.maxNumEventLoopsPerEndpoint(int)>` and
  `<type://ClientFactoryBuilder.maxNumEventLoopsPerHttp1Endpoint(int)>` when a client
  needs to send a large number of requests to a specific endpoint. The client will
  assign more CPU resources and create more connections by increasing the number of
  event loops.
```

> - Increase `maxNumEventLoopsPerEndpoint` and `maxNumEventLoopsPerHttp1Endpoint` when your clients send requests
>   to a small number of servers. As an Endpoint uses a single event loop, there may be a bottleneck.

Reviewer 3: - What do you think of adding `Consider` since performance may not always be better.
- What do you think of conveying why there is a bottleneck?

```suggestion
- Consider increasing `maxNumEventLoopsPerEndpoint` and `maxNumEventLoopsPerHttp1Endpoint` for better request throughput. Doing so may increase performance at the cost of opening more connections.
```

Reviewer 3: I saw this comment late https://github.com/line/armeria/pull/5297#discussion_r1391916506 . 
I think using this comment is fine, but still I think adding "Consider" is a good idea
